src/common/row_operations/row_match.cpp

```diff
@@ -261,4 +261,124 @@ idx_t RowOperations::Match(DataChunk &columns, VectorData col_data[], const RowL
 	return count;
 }
 
+template <class T, class OP>
+static void TemplatedMatchRowsType(Vector &rows_left, const SelectionVector &left_sel, Vector &rows_right,
+                                   const SelectionVector &right_sel, idx_t rows_count, idx_t col_offset,
+                                   idx_t &matches) {
+	// Precompute row_mask indexes
+	idx_t entry_idx;
+	idx_t idx_in_entry;
+	ValidityBytes::GetEntryIndex(col_offset, entry_idx, idx_in_entry);
+
+	auto left_ptrs = FlatVector::GetData<data_ptr_t>(rows_left);
+	auto right_ptrs = FlatVector::GetData<data_ptr_t>(rows_right);
+	idx_t match_count = 0;
+
+	for (idx_t i = 0; i < rows_count; i++) {
+		auto left_idx = left_sel.get_index(i);
+		auto right_idx = right_sel.get_index(i);
+
+		auto left_row = left_ptrs[left_idx];
+		auto right_row = right_ptrs[right_idx];
+		ValidityBytes left_row_mask(left_row);
+		ValidityBytes right_row_mask(right_row);
+		// auto isnull = !left_row_mask.RowIsValid(left_row_mask.GetValidityEntry(entry_idx), idx_in_entry);
+
+		auto right_value = Load<T>(right_row + col_offset);
+		auto left_value = Load<T>(left_row + col_offset);
+		if (OP::template Operation<T>(left_value, right_value)) {
+			match_count++;
+		}
+	}
+
+	matches = match_count;
+}
+
+template <class OP>
+static void TemplatedMatchRowsOp(Vector &rows_left, const SelectionVector &left_sel, const RowLayout &layout,
+                                 const idx_t keys_count, Vector &rows_right, const SelectionVector &right_sel,
+                                 idx_t rows_count, idx_t &matches) {
+	if (rows_count == 0) {
+		return;
+	}
+	auto col_offsets = layout.GetOffsets();
+	auto condition_types = layout.GetTypes();
+	// for each key_type check whether there is a match
+	for (idx_t key_idx = 0; key_idx != keys_count; ++key_idx) {
+		auto key_type = condition_types[key_idx];
+		auto key_offset = col_offsets[key_idx];
+		switch (key_type.InternalType()) {
+		case PhysicalType::BOOL:
+		case PhysicalType::INT8:
+			TemplatedMatchRowsType<int8_t, OP>(rows_left, left_sel, rows_right, right_sel, rows_count, key_offset,
```

Comment:
Most of these types are not covered, could you add tests that cover all of the types here?